library/module_utils/oneview.py

```diff
@@ -1021,8 +1021,14 @@ def _merge_os_deployment_custom_attr(self, merged_data, resource, data):
                 existing_attributes = existing_os_deployment[SPKeys.ATTRIBUTES]
                 params_attributes = params_os_deployment[SPKeys.ATTRIBUTES]
 
-                if compare_list(existing_attributes, params_attributes):
-                    merged_os_deployment[SPKeys.ATTRIBUTES] = existing_attributes
+                merged_data[SPKeys.OS_DEPLOYMENT][SPKeys.ATTRIBUTES] = merge_list_by_key(
+                    existing_attributes,
+                    params_attributes,
+                    key='name',
+                )
```

Comment:
merged_data[SPKeys.OS_DEPLOYMENT][SPKeys.ATTRIBUTES] = merge_list_by_key(
                    params_attributes,
                    existing_attributes,
                    key='name',
                )

Reply:
I have fixed the sp idempotency code. Please review it.